File: frontend/src/types/Job.ts
export type JobStatus =
  | 'Received'
  | 'Pending'
  | 'Processing'
  | 'Printing'
  | 'Done'
  | 'Error'
  | 'Retained'
  | 'AwaitingOperatorConfirmation'
  | 'BlockedSecurityViolation';

export type JobType =
  | 'BankStatement'
  | 'SalaryCheque'
  | 'InsuranceDocument'
  | 'Bill'
  | 'TaxDocument'
  | 'ColorDocument';

export interface JobParameters {
  id: number;
  jobId: number;
  isDuplex: boolean;
  colorMode: string;
  copies: number;
  totalPageCount: number;
  isPayrollJob: boolean;
  employeeCount: number | null;
  pagesPerEmployee: number | null;
  containsCheques: boolean;
  containsStubs: boolean;
  paperType: string;
}

export interface JobHistory {
  id: number;
  jobId: number;
  action: string;
  operatorId: number | null;
  timestamp: string;
  details: string | null;
}

export interface Job {
  id: number;
  jobName: string;
  clientName: string;
  fileLocation: string;
  fileSize: number;
  status: JobStatus;
  jobType: JobType;
  printerId: number | null;
  queueProfileId: number | null;
  operatorId: number | null;
  printerJobId: string | null;
  createdAt: string;
  sentAt: string | null;
  completedAt: string | null;
  printer: import('./Printer').Printer | null;
  queueProfile: import('./Profile').QueueProfile | null;
  parameters: JobParameters | null;
  history: JobHistory[];
}

File: backend/PrintSaaS.Models/Job.cs
using PrintSaaS.Models.Enums;

namespace PrintSaaS.Models;

public class Job
{
    public int Id { get; set; }
    public string JobName { get; set; } = string.Empty;
    public string ClientName { get; set; } = string.Empty;
    public string FileLocation { get; set; } = string.Empty;
    public long FileSize { get; set; }
    public JobStatus Status { get; set; }
    public JobType JobType { get; set; }
    public int? PrinterId { get; set; }
    public int? QueueProfileId { get; set; }
    public int? OperatorId { get; set; }
    public string? PrinterJobId { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime? SentAt { get; set; }
    public DateTime? CompletedAt { get; set; }

    public Printer? Printer { get; set; }
    public QueueProfile? QueueProfile { get; set; }
    public User? Operator { get; set; }
    public JobParameters? Parameters { get; set; }
    public ICollection<JobHistory> History { get; set; } = new List<JobHistory>();
}
